### PR TITLE
math: sign function

### PR DESCRIPTION
**Pull request by cabralski**
This PR adds a new function to `math.v` called `sign(n f64)`, it's a simple function but really essential when constructing other mathematical functions.

A minimalist test `test_sign` was added alongside with the function.

### CODE REVIEW

**Comment by medvednikov** [comment]
Go decided no to implement this: https://github.com/golang/go/issues/3743

This SO answer explaining why: https://stackoverflow.com/questions/1986152/why-doesnt-python-have-a-sign-function

But Java does have it.

I think it's fine, can you also add a `copysign()` function please?

**Comment by cabralski** [comment]
> I think it's fine, can you also add a `copysign()` function please?

There's already a `copysign` at `math.v`, do you mean that I should use `abs(x) * sign(x)` then?

```v
// copysign returns a value with the magnitude of x and the sign of y
pub fn copysign(x f64, y f64) f64 {
	return f64_from_bits((f64_bits(x) & ~sign_mask) | (f64_bits(y) & sign_mask))
}
```

File: vlib/math/math.v [hunk 1]
```diff
@@ -114,6 +114,17 @@ pub fn min(a f64, b f64) f64 {
 	return b
 }
 
+// sign returns the corresponding sign [-1, 0, 1] of the provided number.
+pub fn sign(n f64) int {
+	if n > 0 {
```
Comment (Gladear) [inline vlib/math/math.v:119]: For a more efficient implementation, you can implementation `sign` as suggested in @medvednikov's link
```suggestion
	return copysign(1, n)
```

**Comment by cabralski** [comment]
Oh, I get it now @Gladear! Added your changes.

**Comment by Gladear** [comment]
Yeah, huu, I failed my multi-line suggestion apparently 😅 I only changed one line, sorry about that

**Comment by cabralski** [comment]
No problem! Just fixed it.

**Comment by medvednikov** [comment]
```
vlib/math/math.v:119:9: error: cannot use `f64` as type `int` in return argument
  117 | // sign returns the corresponding sign [-1, 0, 1] of the provided number.
  118 | pub fn sign(n f64) int {
  119 |     return copysign(1, n)
      |            ~~~~~~~~~~~~~~
  120 | }
```

**Review by hungrybluedev — COMMENTED** [review]
Seems fine to me. The unit tests for `sign()` could do with more variety. Like random digit-mashed floats.

**Comment by ulises-jeremias** [comment]
I think we need to add a test for `math.sign(nan())`. In languages with `nan` and `-nan`, it is possible to think that it will return `1` and `-1` respectively. If I am not wrong, we don't have `-nan`, so we should return `nan` for `math.sign(nan())`, so it is the extra edge case that we have to test (and document properly).

**Review by ulises-jeremias — COMMENTED** [review]
I think we need to add a test case for `math.sign(nan())`. In languages with `nan` and `-nan`, it is possible to think that it will return `1` and `-1` respectively. If I am not wrong, we don't have `-nan`, so we should return `nan` for `math.sign(nan())`, so it is the extra edge case that we have to test (and document properly).

In that case we need to change `math.sign` to 

```v
pub fn sign(n f64) f64
```

**Comment by ulises-jeremias** [comment]
Anyway, we could have another function with the actual implementation that just returns an int, knoing that `int(math.copysign(math.nan())) == 1`. That function could be

```v
pub fn signi(n f64) int
```

**Comment by ulises-jeremias** [comment]
So in that case we contemplate both common cases, sign that handle math.nan() and returns nan, and a function that just returns an int. What do you think guys?

**Review by cabralski — COMMENTED** [review]
I think `sign` is good to go!

**Comment by spytheman** [comment]
> Seems fine to me. The unit tests for `sign()` could do with more variety. Like random digit-mashed floats.

done

**Comment by spytheman** [comment]
> ```v
> signi
> ```

done

File: vlib/math/math.v [hunk 2]
```diff
@@ -107,19 +111,34 @@ pub fn max(a f64, b f64) f64 {
...
 	if a < b {
 		return a
 	}
 	return b
 }
 
+// sign returns the corresponding sign [-1.0, 1.0] of the provided number.
+[inline]
+pub fn sign(n f64) f64 {
+	return copysign(1.0, n)
```
Comment (ulises-jeremias) [inline vlib/math/math.v:125]: we need to add a validation here. 

```v
[inline]
pub fn sign(n f64) f64 {
    if is_nan(n) {
        return nan()
    }
	return copysign(1.0, n)
}
```

**Review by ulises-jeremias — CHANGES_REQUESTED** [review]

File: vlib/math/math.v (hunk 2 above)
Comment (ulises-jeremias) [reply vlib/math/math.v:125]: and also a test for that case

Comment (spytheman) [reply vlib/math/math.v:128]: done
